hack/common.sh

```diff
@@ -456,12 +456,12 @@ os::build::ldflags() {
     os::build::get_version_vars
 
     declare -a ldflags=()
-    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.majorFromGit" "${OS_GIT_MAJOR}")
-    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.minorFromGit" "${OS_GIT_MINOR}")
-    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.versionFromGit" "${OS_GIT_VERSION}")
-    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.commitFromGit" "${OS_GIT_COMMIT}")
-    ldflags+=(-X "k8s.io/kubernetes/pkg/version.gitCommit" "${KUBE_GIT_COMMIT}")
-    ldflags+=(-X "k8s.io/kubernetes/pkg/version.gitVersion" "${KUBE_GIT_VERSION}")
+    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.majorFromGit=${OS_GIT_MAJOR}")
+    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.minorFromGit=${OS_GIT_MINOR}")
+    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.versionFromGit=${OS_GIT_VERSION}")
+    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.commitFromGit=${OS_GIT_COMMIT}")
+    ldflags+=(-X "k8s.io/kubernetes/pkg/version.gitCommit=${KUBE_GIT_COMMIT}")
+    ldflags+=(-X "k8s.io/kubernetes/pkg/version.gitVersion=${KUBE_GIT_VERSION}")
```

Comment:
From my tests on s2i, you'll have to conditionally do old version vs this new version.